### PR TITLE
Travis-CI

### LINKED ISSUE #8: travis-ci.org integration
add a .travis.yml for travis-ci integration

### PR DESCRIPTION
**Pull request by ax3l**
Close #8: Enable Travis-CI

This PR adds CI via travis-ci.org . Some minor changes in `requirements.txt` were added to fix some build failures.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+before_script:
+  # print interesting versions
+  - python --version
+  - ipython --version
+  - python -c "import numpy as np; print( np.__version__ )"
+  - python -c "import h5py as h5; print( h5.__version__ )"
+  - python -c "import mpi4py; print( mpi4py.__version__ )"
+  # static code analysis
+  #   pyflakes: mainly warnings, unused code, etc.
+  # - python -m pyflakes .
```
Comment (ax3l) [inline .travis.yml:104]: As soon as the current errors documented in #19 are fixed we comment in this line.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+matrix:
+  include:
+    - env: USE_ADIOS=OFF
+  # - env: USE_ADIOS=SERIAL
+  #   apt:
+  #     packages: libadios-dev
+  # - env: USE_ADIOS=MPI
+  #   apt:
+  #     packages: libadios-dev
+  # - env: USE_ADIOS=OFF USE_CUDA=ON
```
Comment (ax3l) [inline .travis.yml]: disabled, waiting for https://github.com/travis-ci/apt-package-whitelist/issues/2186

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+language: python
+python:
+  - "2.7"
+  - "3.5"
+
+matrix:
+  include:
+    - env: USE_ADIOS=OFF
+  # - env: USE_ADIOS=SERIAL
```
Comment (ax3l) [inline .travis.yml]: disabled, waiting for https://github.com/travis-ci/apt-package-whitelist/issues/2187

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  # - USE_ADIOS=MPI
+  # - USE_ADIOS=OFF USE_CUDA=ON
+
+addons:
+  apt:
+    packages:
+      - openmpi-bin
+      - libopenmpi-dev
+      - libhdf5-openmpi-dev
+    # - libadios-dev
```
Comment (ax3l) [inline .travis.yml]: disabled, waiting for https://github.com/travis-ci/apt-package-whitelist/issues/2187

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    # - nvidia-common
+    # - nvidia-cuda-dev
+    # - nvidia-cuda-toolkit
```
Comment (ax3l) [inline .travis.yml:50]: disabled, waiting for https://github.com/travis-ci/apt-package-whitelist/issues/2186

**Comment by ax3l** [comment]
@Heikman hm, I don't get why the `pyDive/test/test_ndarray.py` test [is taking so long to run](https://travis-ci.org/ComputationalRadiationPhysics/pyDive/jobs/100677555) (Python2.7)

Also, looks like the tests are not all Python3 compatible yet: https://travis-ci.org/ComputationalRadiationPhysics/pyDive/jobs/100677566

**Comment by heikoburau** [comment]
There are five tests for `ndarray`. Seems that it hangs on the second.

**Comment by ax3l** [comment]
> Seems that it hangs on the second.

can you investigate and maybe fix that one?

**Comment by ax3l** [comment]
@Heikman merg-able :)

**Comment by ax3l** [comment]
@Heikman there seems to be a Python3 problem left in `pyDive/ipyParallelClient.py` with `MPI` used in the `lambda` to get the rank:

``` python
    # mpi ranks
    get_rank = interactive(lambda: MPI.COMM_WORLD.Get_rank())
    all_ranks = view.apply(get_rank)
    view['target2rank'] = all_ranks
```

```
E               CompositeError: one or more exceptions from call to method: <lambda>
E               [0:apply]: NameError: global name 'MPI' is not defined
```

is `init` lacking an `@interactive` [decorator](https://stackoverflow.com/questions/10461774/ipython-parallel-namespaces) or something?

also, adios in python2 segfaults for no apparent reason.

**Comment by ax3l** [comment]
offline result: Maybe we can do the imports via [sync_imports](https://ipython.org/ipython-doc/dev/api/generated/IPython.parallel.client.view.html#IPython.parallel.client.view.DirectView.sync_imports) instead

**Comment by heikoburau** [comment]
python3 is now supported in dev. Can you please rebase and try again?
On hypnos adios is working with python2, however there is no python3 support for adios yet.

**Comment by heikoburau** [comment]
still have no idea for the "missing MPI" problem.

**Comment by ax3l** [comment]
@Heikman updated! `test/test_ndarray.py:42 and :85` seems to [have a problem](https://travis-ci.org/ComputationalRadiationPhysics/pyDive/jobs/109157922):

```
pyDive/test/test_ndarray.py FException IndexError: 'list index out of range' in <bound method ndarray.__del__ of dist_array168>
```

**Comment by heikoburau** [comment]
I see, I didn't use the latest flake8 version.

**Comment by heikoburau** [comment]
added a bugfix.

**Comment by ax3l** [comment]
shall I rebase?

**Comment by heikoburau** [comment]
already done

**Comment by ax3l** [comment]
that explains the weird commit meta data ;)

**Comment by heikoburau** [comment]
I'm working on the flake8 issues right now so we have to rebase soon again.

**Comment by ax3l** [comment]
awesome! I guess the problem travis found was e7d2939f528ba91e6336ed18f1b266a6b42d8ca1 ?

**Comment by heikoburau** [comment]
yep, but only after updating to the newest libraries.

**Comment by heikoburau** [comment]
okay, you can enabled the flake8 command again, but now additionally with `--ignore=E226,E731`.

**Comment by ax3l** [comment]
@Heikman rebased and compiles in Python 2 & 3. Adios is still segfaulting, but we can fix this in a follow-up if you want.

**Comment by heikoburau** [comment]
Okay, can you just combine the commits to a single one?

**Comment by ax3l** [comment]
I changed three files so 2 would be a good idea imho?

**Comment by Heikman** [comment]
+1

**Comment by ax3l** [comment]
@Heikman did you remove or even `rebase` / `push --force` to the already published `dev` branch?

the Travis-CI integration from this PR is gone in `dev`.

**Pull request merged**